airbyte-integrations/connectors/source-zuora/source_zuora/zuora_client.py

```diff
@@ -0,0 +1,266 @@
+# copies of the Software, and to permit persons to whom the Software is
+# furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included in all
+# copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+# SOFTWARE.
+#
+
+
+import json as j
+from datetime import datetime, timedelta
+from math import ceil
+from typing import Dict, Iterable, List
+
+import pendulum
+import requests
+from airbyte_cdk import AirbyteLogger
+
+from .zuora_auth import ZuoraAuthenticator
+from .zuora_errors import ZOQLQueryCannotProcessObject, ZOQLQueryFailed, ZOQLQueryFieldCannotResolve
+
+
+class ZoqlExportClient:
+
+    """
+    # TODO: Add the description about the ZOQL EXPORT + Links
+    """
+
+    logger = AirbyteLogger()
+
+    def __init__(self, authenticator: Dict, url_base: str, **config: Dict):
+        self.authenticator = authenticator
+        self.url_base = url_base
+        self.start_date = config["start_date"]
+        self.window_in_days = config["window_in_days"]
+        self.client_id = config["client_id"]
+        self.client_secret = config["client_secret"]
+        self.is_sandbox = config["is_sandbox"]
+        self.retry_max = 3  # Max number of retries for make_request method
+
+    def make_request(self, method: str = "GET", url: str = None, payload: Dict = None) -> requests.Response:
+        """
+        try/except request wraper with handling errors
+        """
+
+        retry = 0
+        while retry <= self.retry_max:
+            try:
+                request = requests.request(method=method, url=url, headers=self.authenticator, json=payload)
+                request.raise_for_status()
+                return request
+            except requests.exceptions.HTTPError as e:
+                retry += 1
+                if retry <= self.retry_max:
+                    self.handle_requests_exceptions(e)
+                    continue
+                else:
+                    raise Exception(e)
+
+    def make_query(
+        self, q_type: str = "select", obj: str = None, date_field: str = None, start_date: str = None, end_date: str = None
+    ) -> Dict:
+        """
+        Make the query for dependent methods like:
+        submit_job, check_job_status, zuora_list_streams, zuora_get_json_schema
+        """
+        # Define base query parameters
+        base_query = {"compression": "NONE", "output": {"target": "S3"}, "outputFormat": "JSON"}
+
+        if q_type == "select":
```

Comment:
each query type - is a separate HttpStream

Reply:
This is ZouraBase stream  - all  streams for another objects will be inherited from this stream